Weather/WeatherWidget.swift

```diff
@@ -9,66 +9,151 @@
+            if Day == 10{
+                Day = 0
+            }else{
+                Day = Day + 1
+            }
+            Wait = Wait + 1
+            NotificationCenter.default.post(name: .didChangeWidgetPreferences, object: nil)
+            DispatchQueue.main.asyncAfter(deadline: .now() + 10) {
+                if Wait == 1{
+                    Day = 0
+                    NotificationCenter.default.post(name: .didChangeWidgetPreferences, object: nil)
+                }
+                Wait = Wait - 1
+            }
+        }
     }
 }
-
 public class WeatherWidget: PKWidget {
     public static var identifier: String = "WeatherWidget"
-    public var customizationLabel: String = "Weather"
-    public var view: NSView!
-    
-    private var weatherRepository: WeatherRepository? = WeatherRepository()
-	private var data: WeatherData?
+        public var customizationLabel: String = "Weather"
+        public var view: NSView!
+        private var weatherRepository: WeatherRepository? = WeatherRepository()
+        private var data: WeatherData?
 
-    required public init() {
-        self.view = WeatherView(leftToRight: false)
-		self.weatherRepository?.set(completionBlock: { [weak self] data in
-			print("[WeatherWidget]: Updated weather data")
-			self?.data = data
-			self?.update()
-		})
+        required public init() {
+            self.view = WeatherView(leftToRight: false)
+            self.weatherRepository?.set(completionBlock: { [weak self] data in
+                print("[WeatherWidget]: Updated weather data")
+                self?.data = data
+            self?.update()
+        })
         NotificationCenter.default.addObserver(self, selector: #selector(update), name: .didChangeWidgetLayout, object: nil)
     }
     
     deinit {
         NotificationCenter.default.removeObserver(self)
-		print("[WeatherWidget]: Deinit")
+        print("[WeatherWidget]: Deinit")
         weatherRepository = nil
         view = nil
-		data = nil
+        data = nil
     }
     
     @objc private func update() {
         guard let view = view as? WeatherView, let data = data else {
             return
         }
-        view.maxWidth = 120
-        let locality = data.weather.name
-        view.set(title: locality)
-        if Preferences[.show_description] && data.weather.temp > -999 {
-            view.set(subtitle: "\(data.weather.temperature), \(data.weather.description)")
-        } else {
-            view.set(subtitle: data.weather.temperature)
+        if Day == 0{
+            view.maxWidth = 120
+        }else{
+            view.maxWidth = 180
         }
-        if let localIcon = Bundle(for: Self.self).image(forResource: data.weather.icon) {
-            view.set(image: localIcon)
-        } else if let systemIcon = NSImage(named: data.weather.icon) {
-            view.set(image: systemIcon)
+        let locality = data.weather.name
+        if Preferences[.ShowIconOnly] == false || !(Day == 0){
```

Comment:
Shows icon only based on user settings. Intentionally shows information if not viewing current forecast.